pb-commit-message-lints/src/lints/lib/commit_message.rs

```diff
@@ -0,0 +1,259 @@
+use regex::Regex;
+use std::{convert::TryFrom, fmt::Display, fs::File, io::Read, path::PathBuf};
+
+use crate::errors::PbCommitMessageLintsError;
+
+#[derive(Debug, PartialEq)]
+pub struct CommitMessage {
+    contents: String,
+    comment_char: String,
+}
+
+impl CommitMessage {
+    #[must_use]
+    pub fn new(contents: String) -> CommitMessage {
+        let comment_char = "#".into();
+        CommitMessage {
+            contents,
+            comment_char,
+        }
+    }
+
+    pub fn matches_pattern(&self, re: &Regex) -> bool {
+        re.is_match(&self.contents)
+    }
+
+    #[must_use]
+    pub fn get_trailer(&self, trailer: &str) -> Vec<&str> {
+        self.contents
+            .lines()
+            .filter(|line: &&str| CommitMessage::line_has_trailer(trailer, line))
+            .collect::<Vec<_>>()
+    }
+
+    #[must_use]
+    pub fn add_trailer(&self, trailer: &str) -> Self {
```

Comment:
Can you talk me through the intention of this?

Reply:
It takes the trailer string and it returns a new `CommitMessage` with the contents updated to include the new trailer in the right place.

It does this by splitting the message into 2 parts, the main message content and the trailing comment. It them puts them back together with the trailer line in-between.

Have I answered your question or have I missed what you were asking?

Reply:
Nah, I was just wondering about if it was used outside tests mostly

Reply:
It's used here https://github.com/PurpleBooth/pb-git-hooks/blob/master/pb-prepare-commit-msg/src/main.rs#L92